src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager.cs

```diff
@@ -0,0 +1,216 @@
+        ApplyTheme(new ArrayList { window } , systemTheme, useLightMode, systemAccentColor, forceUpdate);
+    }
+
+    internal static void ApplyTheme(
+        IEnumerable windows, 
+        string requestedTheme, 
+        bool requestedUseLightMode,
+        Color requestedAccentColor, 
+        bool forceUpdate = false)
+    {
+        bool needsUpdate = forceUpdate;
+        
+        if(requestedUseLightMode != _currentUseLightMode || DwmColorization.GetSystemAccentColor() != DwmColorization.CurrentApplicationAccentColor)
+        {
+            DwmColorization.UpdateAccentColors();
+            needsUpdate = true;
+        }
+
+        if(needsUpdate || requestedTheme != _currentApplicationTheme || requestedUseLightMode != _currentUseLightMode)
+        {
+            Uri dictionaryUri = GetFluentWindowThemeResourceUri(requestedTheme, requestedUseLightMode);
+
+            UpdateFluentWindowsThemeResources(dictionaryUri);
+
+            foreach(Window window in windows)
+            {
+                SetImmersiveDarkMode(window, !requestedUseLightMode);
+                window.CoerceValue(Window.WindowBackdropTypeProperty);
+            }
+
+            _currentApplicationTheme = requestedTheme;
+            _currentUseLightMode = requestedUseLightMode;
+        }
+    }
+
+    private static bool SetImmersiveDarkMode(Window window, bool useDarkMode)
+    {
+        if (window == null)
+        {
+            return false;
+        }
+
+        IntPtr handle = new WindowInteropHelper(window).Handle;
+
+        if (handle != IntPtr.Zero)
+        {
+            var dwmResult = NativeMethods.DwmSetWindowAttributeUseImmersiveDarkMode(handle, useDarkMode);
+            return dwmResult == HRESULT.S_OK;
+        }
+
+        return false;
+    }
+
+    #region Helper Methods
+
+    internal static string GetSystemTheme()
+    {
+        string systemTheme = Registry.GetValue(_regThemeKeyPath,
+            "CurrentTheme", "aero.theme") as string ?? String.Empty;
+
+        return systemTheme;
+    }
+   
+    internal static bool GetUseLightTheme()
+    {
+        var appsUseLightTheme = Registry.GetValue(_regPersonalizeKeyPath,
+            "AppsUseLightTheme", null) as int?;
+
+        if (appsUseLightTheme == null)
+        {
+            return Registry.GetValue(_regPersonalizeKeyPath,
+                "SystemUsesLightTheme", null) as int? == 0 ? false : true;
+        }
+
+        return appsUseLightTheme != 0;
+    }
+
+    internal static void UpdateFluentWindowsThemeResources(Uri dictionaryUri)
+    {
+        ArgumentNullException.ThrowIfNull(dictionaryUri, nameof(dictionaryUri));
+
+        var newDictionary = new ResourceDictionary() { Source = dictionaryUri };
+
+        foreach (var key in newDictionary.Keys)
+        {
+            if (Application.Current.Resources.Contains(key))
+            {
+                if (!object.Equals(Application.Current.Resources[key], newDictionary[key]))
```

Comment:
The ResourceDictionary checks for the equivalence itself, you can save doing it here.